packages/samples/headless-react/src/components/date-facet/date-facet.class.tsx

```diff
@@ -0,0 +1,95 @@
+import {Component} from 'react';
+import {
+  buildDateFacet,
+  DateFacet as HeadlessDateFacet,
+  DateFacetOptions,
+  DateFacetState,
+  DateFacetValue,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+import {parseDate} from './date-utils';
+
+type DateRangeRequest = Required<DateFacetOptions>['currentValues'][number]; // TODO: Remove this when using the new interface.
+
+interface BaseDateFacetProps {
+  field: string;
+  facetId: string;
+}
+
+interface AutomaticDateFacetProps extends BaseDateFacetProps {
+  generateAutomaticRanges: true;
+  currentValues?: DateRangeRequest[];
+}
+
+interface ManualDateFacetProps extends BaseDateFacetProps {
+  generateAutomaticRanges: false;
+  currentValues: DateRangeRequest[];
+}
```

Comment:
Same comment as the NumericFacet sample on not including these interfaces in the sample in favour of using the props interface exported by headless.

Reply:
> The common pattern with the samples so far was to redefine the interfaces since *Props interfaces typically contain `initialState`, `options` and nothing else, so for now I'll just remove the discriminating union.
>
> However, we could create a story after the samples are done to change them all to use the *Props or *Options interfaces.